File: irrgartenjaava/src/irrgartenjaava/GameState.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgartenjaava;

/**
 *
 * @author jxlig0d
 */
public class GameState {
    
    private String labyrinth;
    private String players;
    private String monsters;
    private int currentPlayer;
    private boolean winner;
    private String log;
    
    //Constructor
    
    public GameState(String labyrinth, String players, String monsters, int currentPlayer, boolean winner, String log){
        this.labyrinth = labyrinth;
        this.players = players;
        this.monsters = monsters;
        this.currentPlayer = currentPlayer;
        this.winner = winner;
        this.log = log;
        
    }
    
    // Consultor para labyrinth
    public String getLabyrinth() {
        return labyrinth;
    }

    // Consultor para players
    public String getPlayers() {
        return players;
    }

    // Consultor para monsters
    public String getMonsters() {
        return monsters;
    }

    // Consultor para currentPlayer
    public int getCurrentPlayer() {
        return currentPlayer;
    }

    // Consultor para winner
    public boolean isWinner() {
        return winner;
    }

    // Consultor para log
    public String getLog() {
        return log;
    }
    
    
}

File: irrgarten_ruby/src/game_state.rb
#encoding:utf-8

module Irrgarten

  class Game_state 

    attr_accessor :labyrinth, :players, :monsters, :currentPlayer, :winner, :log

    def initialize(lab, player, monster, ctPlayer, winner_, log_)
      @labyrinth = lab
      @players = player
      @monsters = monster
      @currentPlayer = ctPlayer
      @winner = winner_
      @log = log_
    end

    def get_labyrinth
      return @labyrinth
    end

    def get_players
      return @players
    end

    def get_monsters
      return @monsters
    end

    def get_currentPlayer
      return @currentPlayer
    end

    def is_winner
      return @winner
    end

    def get_log
      return @log
    end

  end #end class
end #end module 